### PR TITLE
use in process clicks by default

### PR DESCRIPTION
**Pull request by dimorinny**

### CODE REVIEW

File: test-app/src/main/res/layout/activity_moving_button.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    android:layout_height="match_parent">
+
+    <View
+        android:id="@+id/death_zone"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:background="@color/black" />
+
+    <Button
+        android:id="@+id/moved_button"
```
Comment (dsvoronin) [inline test-app/src/main/res/layout/activity_moving_button.xml]: moving

File: test-app/src/main/java/com/avito/android/ui/Util.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.avito.android.ui
+
+import android.view.View
+
+fun View.toggleVisibility() {
```
Comment (dsvoronin) [inline test-app/src/main/java/com/avito/android/ui/Util.kt]: Views.kt

File: test-app/src/main/java/com/avito/android/ui/MovingButtonActivity.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_moving_button)
+
+        val deathZone = findViewById<View>(R.id.death_zone)
+        val movedButton = findViewById<Button>(R.id.moved_button)
+        val movedButtonClickedIndicator = findViewById<TextView>(R.id.moved_button_clicked_text_view)
+
+        deathZone.setOnClickListener { throw RuntimeException("Clicked to a death zone") }
```
Comment (dsvoronin) [inline test-app/src/main/java/com/avito/android/ui/MovingButtonActivity.kt]: Clicked on

File: ui-testing-core/src/main/java/com/avito/android/test/UITestConfig.kt [hunk 4]
```diff
@@ -43,15 +43,40 @@ object UITestConfig {
...
+                object TryOneMoreClick : ClickRollbackPolicy()
+                object Fail : ClickRollbackPolicy()
+            }
+        }
+
+        /**
+         * Use clicks and long clicks implementation without IPC.
+         *
+         * Advantages:
+         *  - Can click to moving objects (espresso has problems with clicking to moving objects, because
```
Comment (dsvoronin) [inline ui-testing-core/src/main/java/com/avito/android/test/UITestConfig.kt]: click on; clicking on

File: ui-testing-core/src/main/java/com/avito/android/test/UITestConfig.kt [hunk 5]
```diff
@@ -43,15 +43,40 @@ object UITestConfig {
...
+         *    it takes some time to touch after coordinates calculating).
+         *  - Doesn't have problems with clicks that can be interpreted as long clicks.
```
Comment (dsvoronin) [inline ui-testing-core/src/main/java/com/avito/android/test/UITestConfig.kt]: Doesn't have "misinterpret clicks as long clicks" problem.

File: ui-testing-core/src/main/java/com/avito/android/test/espresso/LibraryViewActions.kt [hunk 6]
```diff
@@ -63,4 +70,42 @@ object EspressoActions {
             )
         )
     }
+
+    fun click(type: UITestConfig.ClickType = UITestConfig.clicksType): ViewAction = ActionOnEnabledElement(
+        when (type) {
+            is UITestConfig.ClickType.EspressoClick -> when (type.rollbackPolicy) {
+
+                is UITestConfig.ClickType.EspressoClick.ClickRollbackPolicy.DoNothing -> ViewActions.click()
```
Comment (dsvoronin) [inline ui-testing-core/src/main/java/com/avito/android/test/espresso/LibraryViewActions.kt]: static imports would be helpful here

**Review by dsvoronin — DISMISSED** [review]

File: ui-testing-core/src/main/java/com/avito/android/test/UITestConfig.kt (hunk 4 above)
Comment (eugene-krivobokov) [inline ui-testing-core/src/main/java/com/avito/android/test/UITestConfig.kt]: `click on moving ...` or just `click moving ...`

Comment (dimorinny) [reply ui-testing-core/src/main/java/com/avito/android/test/UITestConfig.kt]: duplicate

File: ui-testing-core/src/main/java/com/avito/android/test/espresso/LibraryViewActions.kt (hunk 6 above)
Comment (dimorinny) [reply ui-testing-core/src/main/java/com/avito/android/test/espresso/LibraryViewActions.kt]: don't like it(

**Review by eugene-krivobokov — APPROVED** [review]

**Pull request merged**